src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/convenience/key_vault_credential.py

```diff
@@ -7,18 +7,53 @@
 
 import adal
 import urllib
+from requests.auth import AuthBase
+import requests
 
 from msrest.authentication import Authentication
 
 from . import HttpBearerChallenge
 from . import HttpBearerChallengeCache as ChallengeCache
 
-class KeyVaultCredential(Authentication):
+class KeyVaultAuthBase(AuthBase):
 
-    def __init__(self, token):
-        self._token = token
+    def __init__(self, token_callback):
```

Comment:
Since the code will get into the library, let us fill in doc-string across, add unit tests to have 90%+ coverage.
We can do it when submit PR to python-sdk repo.